packages/edit-site/src/components/page-library/patterns-list.js

```diff
@@ -34,13 +45,32 @@ export default function PatternsList( { categoryId, label, type } ) {
 
 	return (
 		<VStack spacing={ 6 }>
-			<SearchControl
-				className="edit-site-library__search"
-				onChange={ ( value ) => setFilterValue( value ) }
-				placeholder={ __( 'Search patterns' ) }
-				value={ filterValue }
-				__nextHasNoMarginBottom
-			/>
+			<Flex>
+				{ isMobileViewport && (
+					<SidebarButton
+						icon={ isRTL() ? chevronRight : chevronLeft }
+						label={ __( 'Back' ) }
+						onClick={ () => {
+							// Go back in history if we came from the library page.
+							// Otherwise push a stack onto the history.
+							if ( location.state?.backPath === '/library' ) {
+								history.back();
+							} else {
+								history.push( { path: '/library' } );
+							}
```

Comment:
The reason we have to do this instead of leveraging `NavigatorToParentButton` is because this tree isn't wrapped in the `NavigatorProvider` context, nor is it under the `/library` subtree of the `NavigatorScreen`, so `navigator.goBack()` won't work here.

I couldn't figure out a way to do this with the Navigator approach so I fallback to the regular browser's history approach. The way it works is that we can attach a `state` when pushing a stack onto the history. In this case the state is set when clicking category links in the library sidebar. If the current location has the state then we can just call `history.back()` and go back to the library page. Otherwise, when there's no such state (like a reload) then we just push a new entry to the history instead.